### ISSUE TITLE
Fix compile time for A*B

### ISSUE THREAD

**Issue by dlfivefifty** [JuliaMatrices/BandedMatrices.jl#32]
@randy3k There's been a huge slowdown in compiling multiplication:

```julia
julia> using BandedMatrices

julia> A = brand(10,10,0,2);

julia> B = brand(10,10,1,1);

julia> @time A*B
  6.308210 seconds (10.99 M allocations: 435.596 MiB, 3.14% gc time)
10×10 BandedMatrices.BandedMatrix{Float64}:
 0.12743   0.650602   0.640555   0.433851  …                               
 0.196636  0.323239   0.38833    0.317936                                  
           0.0537917  0.142114   0.606653                                  
                      0.0791999  0.680316                                  
                                 0.334344     0.0718981                    
                                           …  0.130922   0.020899          
                                              0.983563   0.336551  0.309136
                                              0.84535    0.779808  0.617574
                                              0.324883   0.25855   0.586666
                                                         0.135284  0.587271
```
In BandedMatrices v0.2.3 this was `0.093452 seconds (7.56 k allocations: 449.235 KiB)`

**Comment by dlfivefifty** [comment]
Ahh... this is just because precompile has been removed

**Comment by dlfivefifty** [comment]
Julia v0.7 it's even worse!
```julia
julia> @time A*B
 25.503650 seconds (51.05 M allocations: 4.222 GiB, 3.36% gc time)
10×10 BandedMatrix{Float64}:
 0.497877   0.456532   0.266529  0.0523235  0.0        0.0        0.0        0.0       0.0        0.0     
 0.0975442  0.490095   0.319046  0.386419   0.0642621  0.0        0.0        0.0       0.0        0.0     
 0.0        0.0986654  0.291876  0.892286   1.15614    0.693963   0.0        0.0       0.0        0.0     
 0.0        0.0        0.17526   0.542259   1.16371    0.50444    0.586751   0.0       0.0        0.0     
 0.0        0.0        0.0       0.222954   0.265389   0.434848   0.30255    0.560964  0.0        0.0     
 0.0        0.0        0.0       0.0        0.230327   0.108402   0.752222   0.878209  0.550925   0.0     
 0.0        0.0        0.0       0.0        0.0        0.0397981  0.218159   0.567882  0.636128   0.114696
 0.0        0.0        0.0       0.0        0.0        0.0        0.0818265  0.187449  0.515231   0.271808
 0.0        0.0        0.0       0.0        0.0        0.0        0.0        0.543648  0.653326   0.838414
 0.0        0.0        0.0       0.0        0.0        0.0        0.0        0.0       0.0459432  0.149745
```

**Comment by dlfivefifty** [comment]
Ref https://github.com/JuliaLang/julia/issues/27512

**Comment by dlfivefifty** [comment]
This is now fixed in Julia v0.7 master!
